### PR TITLE
add an option to disable email canon for LISTOWNMAIL comparisons

### LINKED ISSUE #742: add option to turn off email canon on LISTOWNMAIL
custom email canonicalizers might make choices (like stripping `+subaddr` for all providers) that could cause addresses to incorrectly overlap, and canonicalized `LISTOWNMAIL` could expose `username:email` mappings that people shouldn't actually be able to see, so it would be nice to be able to optionally disable email canonicalization in `LISTOWNMAIL`.

### PR DESCRIPTION
**Pull request by jesopo**
closes #742.

stole modular setting from `pwquality_warn_only`. I'm not sure I like how visually complicated the `if` is now. pls be gentle

### CODE REVIEW

File: modules/nickserv/listownmail.c [hunk 1]
```diff
@@ -75,7 +75,7 @@ mod_init(struct module *const restrict m)
 	MODULE_TRY_REQUEST_DEPENDENCY(m, "nickserv/main")
 
 	service_named_bind_command("nickserv", &ns_listownmail);
-	add_bool_conf_item("LISTOWNMAIL_DONT_CANON", &nicksvs.me->conf_table, 0, &listownmail_dont_canon, false);
+	add_bool_conf_item("LISTOWNMAIL_DONT_CANON", &nicksvs.me->conf_table, 0, &listownmail_canon, true);
```
Comment (ilbelkyr) [inline modules/nickserv/listownmail.c:78]: forgot to rename the option here, looks good otherwise i'd say

**Review by ilbelkyr — CHANGES_REQUESTED** [review]

**Review by ilbelkyr — APPROVED** [review]

File: modules/nickserv/listownmail.c [hunk 2]
```diff
@@ -72,6 +75,7 @@ mod_init(struct module *const restrict m)
 	MODULE_TRY_REQUEST_DEPENDENCY(m, "nickserv/main")
 
 	service_named_bind_command("nickserv", &ns_listownmail);
+	add_bool_conf_item("LISTOWNMAIL_CANON", &nicksvs.me->conf_table, 0, &listownmail_canon, true);
```
Comment (aaronmdjones) [inline modules/nickserv/listownmail.c:78]: This needs a corresponding `del_conf_item()` in `mod_deinit()`.

**Review by aaronmdjones — CHANGES_REQUESTED** [review]

**Review by alyx — APPROVED** [review]
lgtm

(Well, after the existing comment from Aaron is resolved)

File: modules/nickserv/listownmail.c (hunk 2 above)
Comment (jesopo) [reply modules/nickserv/listownmail.c:78]: ( ⌒‿⌒)

**Review by aaronmdjones — APPROVED** [review]

**Pull request merged**